### PR TITLE
planner: fix tiflash cannot find generated column

### LINKED ISSUE #40663: ERROR 1105 (HY000): DB::String TiDB::TableInfo::getColumnName(const DB::ColumnID) const: Invalidate column id 35 for table t_224
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
step1: create table 
```
CREATE TABLE github_events (
 id bigint(20) NOT NULL DEFAULT '0',
 type varchar(29) NOT NULL DEFAULT 'Event',
 created_at datetime NOT NULL DEFAULT '1970-01-01 00:00:00',
 repo_id bigint(20) NOT NULL DEFAULT '0',
 repo_name varchar(140) NOT NULL DEFAULT '',
 actor_id bigint(20) NOT NULL DEFAULT '0',
 actor_login varchar(40) NOT NULL DEFAULT '',
 language varchar(26) NOT NULL DEFAULT '',
 additions bigint(20) NOT NULL DEFAULT '0',
 deletions bigint(20) NOT NULL DEFAULT '0',
 action varchar(11) NOT NULL DEFAULT '',
 number int(11) NOT NULL DEFAULT '0',
 commit_id varchar(40) NOT NULL DEFAULT '',
 comment_id bigint(20) NOT NULL DEFAULT '0',
 org_login varchar(40) NOT NULL DEFAULT '',
 org_id bigint(20) NOT NULL DEFAULT '0',
 state varchar(6) NOT NULL DEFAULT '',
 closed_at datetime NOT NULL DEFAULT '1970-01-01 00:00:00',
 comments int(11) NOT NULL DEFAULT '0',
 pr_merged_at datetime NOT NULL DEFAULT '1970-01-01 00:00:00',
 pr_merged tinyint(1) NOT NULL DEFAULT '0',
 pr_changed_files int(11) NOT NULL DEFAULT '0',
 pr_review_comments int(11) NOT NULL DEFAULT '0',
 pr_or_issue_id bigint(20) NOT NULL DEFAULT '0',
 event_day date NOT NULL,
 event_month date NOT NULL,
 event_year int(11) NOT NULL,
 push_size int(11) NOT NULL DEFAULT '0',
 push_distinct_size int(11) NOT NULL DEFAULT '0',
 creator_user_login varchar(40) NOT NULL DEFAULT '',
 creator_user_id bigint(20) NOT NULL DEFAULT '0',
 pr_or_issue_created_at datetime NOT NULL DEFAULT '1970-01-01 00:00:00',
 KEY index_github_events_on_id (id),
 KEY index_github_events_on_created_at (created_at),
 KEY index_github_events_on_repo_id_type_action_month_actor_login (repo_id,type,action,event_month,actor_login),
 KEY index_ge_on_repo_id_type_action_pr_merged_created_at_add_del (repo_id,type,action,pr_merged,created_at,additions,deletions),
 KEY index_ge_on_creator_id_type_action_merged_created_at_add_del (creator_user_id,type,action,pr_merged,created_at,additions,deletions),
 KEY index_ge_on_actor_id_type_action_created_at_repo_id_commits (actor_id,type,action,created_at,repo_id,push_distinct_size),
 KEY index_ge_on_repo_id_type_action_created_at_number_pdsize_psize (repo_id,type,action,created_at,number,push_distinct_size,push_size),
 KEY index_ge_on_repo_id_type_action_created_at_actor_login (repo_id,type,action,created_at,actor_login),
 KEY index_ge_on_repo_name_type (repo_name,type),
 KEY index_ge_on_actor_login_type (actor_login,type),
 KEY index_ge_on_org_login_type (org_login,type),
 KEY index_ge_on_language (language),
 KEY index_ge_on_org_id_type (org_id,type),
 KEY index_ge_on_repo_name_lower ((lower(repo_name))),
 KEY index_ge_on_language_lower ((lower(language)))
  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
  PARTITION BY LIST COLUMNS(type)
  (PARTITION push_event VALUES IN ('PushEvent'),
   PARTITION create_event VALUES IN ('CreateEvent'),
   PARTITION pull_request_event VALUES IN ('PullRequestEvent'),
   PARTITION watch_event VALUES IN ('WatchEvent'),
   PARTITION issue_comment_event VALUES IN ('IssueCommentEvent'),
   PARTITION issues_event VALUES IN ('IssuesEvent'),
   PARTITION delete_event VALUES IN ('DeleteEvent'),
   PARTITION fork_event VALUES IN ('ForkEvent'),
   PARTITION pull_request_review_comment_event VALUES IN ('PullRequestReviewCommentEvent'),
   PARTITION pull_request_review_event VALUES IN ('PullRequestReviewEvent'),
   PARTITION gollum_event VALUES IN ('GollumEvent'),
   PARTITION release_event VALUES IN ('ReleaseEvent'),
   PARTITION member_event VALUES IN ('MemberEvent'),
   PARTITION commit_comment_event VALUES IN ('CommitCommentEvent'),
   PARTITION public_event VALUES IN ('PublicEvent'),
   PARTITION gist_event VALUES IN ('GistEvent'),
   PARTITION follow_event VALUES IN ('FollowEvent'),
   PARTITION event VALUES IN ('Event'),
   PARTITION download_event VALUES IN ('DownloadEvent'),
   PARTITION team_add_event VALUES IN ('TeamAddEvent'),
   PARTITION fork_apply_event VALUES IN ('ForkApplyEvent'))
```
step2: set tiflash replica
```
alter table github_events set tiflash replica 1
```
step3: create index （make sure this is an isolated step after step2）
```
alter table github_events add index `index_ge_on_actor_login_lower` ((lower(`actor_login`)))
```
step4: load stats
```
load stats 'stats path'
```
too big to upload....

query
```
MySQL [gharchive_dev]> SELECT type, created_at, actor_login, action, number, repo_name FROM github_events WHERE LOWER(actor_login) = LOWER('NGAUT') ORDER BY created_at DESC LIMIT 1;
ERROR 1105 (HY000): DB::String TiDB::TableInfo::getColumnName(const DB::ColumnID) const: Invalidate column id 35 for table t_224
```

### 2. What did you expect to see? (Required)
No error

### 3. What did you see instead (Required)
Decode DAG error

### 4. What is your TiDB version? (Required)

v6.5.0 and master

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #40663

Problem Summary: If you create generated column index after create table, planner may add this index into TableScan, which will be pushdown tiflash. But tiflash doesn't know this index column. So got `cannot find column` error.

### What is changed and how it works?
For tiflash path, add `GeneratedColumn` flag to column info, then tiflash will add an empty placeholder for this column. 
Check https://github.com/pingcap/tiflash/pull/6796
### Check List

Tests 

- [ ] Unit test
- [x] Integration test: check https://github.com/pingcap/tiflash/pull/6796/files#diff-4bbadf955e9625b3c29acc68462ca087850b99b0497261818d306e265c09cee4
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/mpp_gather.go [hunk 1]
```diff
@@ -159,7 +166,14 @@ func (e *MPPGather) Open(ctx context.Context) (err error) {
 // Next fills data into the chunk passed by its caller.
 func (e *MPPGather) Next(ctx context.Context, chk *chunk.Chunk) error {
 	err := e.respIter.Next(ctx, chk)
-	return errors.Trace(err)
+	if err != nil {
+		return err
+	}
+	err = table.FillVirtualColumnValue(e.virtualColumnRetFieldTypes, e.virtualColumnIndex, e.schema.Columns, e.columns, e.ctx, chk)
```
Comment (windtalker) [inline executor/mpp_gather.go:172]: I'm not sure if this `FillVirtualColumnValue` can work correctly if the mpp request is complex, for example assuming that virutal_col1 is defined as a + b in table t2, will the current pr support `select t1.a, t1.b, t2.virual_col1, t2.a, t2.b from t1 join t2 where xxx`?

File: executor/tiflashtest/tiflash_test.go [hunk 2]
```diff
@@ -1425,3 +1425,38 @@ func TestMPPMemoryTracker(t *testing.T) {
 	require.NotNil(t, err)
 	require.True(t, strings.Contains(err.Error(), "Out Of Memory Quota!"))
 }
+
+func TestDisaggregatedTiFlashGenColumn(t *testing.T) {
```
Comment (windtalker) [inline executor/tiflashtest/tiflash_test.go:1429]: This test is only used to test that if a table contains virtual column, TiDB will still generate a MPP plan if the virtual column is not used?

**Review by windtalker — APPROVED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**